RotatedSortedArray.java

```diff
@@ -0,0 +1,115 @@
+// Time Complexity : O(log n), where n is the size of the array 
+// Space Complexity : O(1) // since it's a finite sized array
```

Comment:
The space complexity should be O(1). 
Your reasoning here is incorrect, it is because no additional space apart from the pointers was used.